glusterd2/commands/volumes/volume-reset.go

```diff
@@ -0,0 +1,111 @@
+package volumecommands
+
+import (
+	"net/http"
+
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	"github.com/gluster/glusterd2/glusterd2/peer"
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/pkg/api"
+	"github.com/gluster/glusterd2/pkg/errors"
+
+	"github.com/gorilla/mux"
+	"github.com/pborman/uuid"
+)
+
+func registerVolResetStepFuncs() {
+	var sfs = []struct {
+		name string
+		sf   transaction.StepFunc
+	}{
+		{"vol-reset.Store", storeVolume},
+		{"vol-reset.NotifyVolfileChange", notifyVolfileChange},
+	}
+	for _, sf := range sfs {
+		transaction.RegisterStepFunc(sf.sf, sf.name)
+	}
+}
+
+func volumeResetHandler(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	volname := mux.Vars(r)["volname"]
+	volinfo, err := volume.GetVolume(volname)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, errors.ErrVolNotFound.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	var req api.VolOptionReq
+	if err := restutils.UnmarshalRequest(r, &req); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusUnprocessableEntity, errors.ErrJSONParsingFailed.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	txn := transaction.NewTxn(ctx)
+	for _, k := range req.Options {
+		if _, ok := volinfo.Options[k]; ok {
+			err = txn.Ctx.Delete(k)
+			if err != nil {
+				logger.WithError(err).Error("Option trying to reset is not set or invalid option")
+			}
+		}
+	}
+	defer txn.Cleanup()
+
+	lock, unlock, err := transaction.CreateLockSteps(volinfo.Name)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	allNodes, err := peer.GetPeerIDs()
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	txn.Steps = []*transaction.Step{
+		lock,
+		{
+			DoFunc: "vol-reset.Store",
+			Nodes:  []uuid.UUID{gdctx.MyUUID},
+		},
+		{
+			DoFunc: "vol-reset.NotifyVolfileChange",
+			Nodes:  allNodes,
+		},
+		unlock,
+	}
+
+	for k, v := range req.Options {
```

Comment:
why this is required?

Reply:
This part is where we Re- set the options with new values?